File: frontend/src/app/data/CrysisDto.ts
import {DetrimentToken} from './DetrimentToken';
import {CrysisCard} from './CrysisCard';
import {CrisisDummyCard} from './CrisisDummyCard';

export class CrysisDto {
  constructor(detrimentTokens: DetrimentToken[],
              openedCards: CrysisCard[],
              cardToTokensCount: Map<number, number>,
              forbiddenPhases: Map<string, number>,
              currentDummyCards: CrisisDummyCard[],
              chosenDummyPhases: number[],
              wonGame: boolean) {
  }

  detrimentTokens: DetrimentToken[];
  openedCards: CrysisCard[];
  cardToTokensCount: Map<number, number>;
  forbiddenPhases: Map<string, number>;
  currentDummyCards: CrisisDummyCard[];
  chosenDummyPhases: number[];
  wonGame: boolean;
}

File: src/main/java/com/terraforming/ares/dto/CrysisDto.java
package com.terraforming.ares.dto;

import com.terraforming.ares.model.CrisisDummyCard;
import com.terraforming.ares.model.DetrimentToken;
import lombok.Builder;
import lombok.Data;

import java.util.*;

/**
 * Created by oleksii.nikitin
 * Creation date 03.03.2023
 */
@Data
@Builder
public class CrysisDto {
    private Set<DetrimentToken> detrimentTokens;
    private List<CrysisCardDto> openedCards;
    private Map<Integer, Integer> cardToTokensCount;
    private Map<String, Integer> forbiddenPhases;
    private List<CrisisDummyCard> currentDummyCards;
    private List<Integer> chosenDummyPhases;
    private boolean wonGame;
}
